### PR TITLE
[#7603] Use enum instead of string for gender field in StudentProfileAttributes and related classes

### LINKED ISSUE #7603: Use enum instead of string for gender field in StudentProfileAttributes and related classes
Replace 'gender' field with corresponding enum for StudentProfileAttributes and related classes.
Currently gender field is String that only accepts "male", "female" or "other" values, so It's better to use enum over String value in that case.

### PR DESCRIPTION
**Pull request by amrut-prabhu**
Fixes #7603  

**Outline of Solution**

Changed the gender constants (Male, Female, Other) to an inner enum `Gender`. Updated affected tests.

Updated json files using this [script](https://github.com/TEAMMATES/teammates/files/2850643/updateGenderInJson.zip)

### CODE REVIEW

File: src/main/java/teammates/common/datatransfer/attributes/Gender.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package teammates.common.datatransfer.attributes;
+
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * Represents the gender of a student.
+ */
+public enum Gender {
+    @SerializedName("male")
```
Comment (xpdavid) [inline src/main/java/teammates/common/datatransfer/attributes/Gender.java]: Do you really need this?

File: src/main/java/teammates/common/datatransfer/attributes/Gender.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package teammates.common.datatransfer.attributes;
+
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * Represents the gender of a student.
+ */
+public enum Gender {
```
Comment (xpdavid) [inline src/main/java/teammates/common/datatransfer/attributes/Gender.java]: Add as inner enum in `StudentProfileAttributes`?

File: src/main/java/teammates/storage/entity/StudentProfile.java [hunk 3]
```diff
@@ -34,8 +36,7 @@
 
     private String nationality;
 
-    /* only accepts "male", "female" or "other" */
```
Comment (xpdavid) [inline src/main/java/teammates/storage/entity/StudentProfile.java]: This is not back-compatible. Maybe you can just leave entity as there and change `StudentProfileAttribute` only.

File: src/main/java/teammates/common/datatransfer/attributes/Gender.java (hunk 1 above)
Comment (amrut-prabhu) [reply src/main/java/teammates/common/datatransfer/attributes/Gender.java]: Hmm some tests were failing locally, but they seem to work fine on Travis. I'll remove it.

File: src/main/java/teammates/common/datatransfer/attributes/StudentProfileAttributes.java [hunk 4]
```diff
@@ -192,8 +214,10 @@ public Builder withNationality(String nationality) {
             return this;
         }
 
-        public Builder withGender(String gender) {
-            profileAttributes.gender = isGenderValid(gender) ? gender : "other";
+        public Builder withGender(Gender gender) {
```
Comment (amrut-prabhu) [inline src/main/java/teammates/common/datatransfer/attributes/StudentProfileAttributes.java]: @xpdavid Would it be better to take a `String` as parameter here instead and then do`profileAttributes.gender = getGenderEnumValue(gender)` ?

**Comment by amrut-prabhu** [comment]
@xpdavid Ready for review

**Comment by xpdavid** [comment]
> Updated json files using this [script]

For text replace, it is doable already in IDE :P. I am worried about accident replacements. Is there any?

I think the correct way is to load as a json object and examine the `gender` filed in `studentProfiles` section.

File: src/main/java/teammates/common/datatransfer/attributes/StudentProfileAttributes.java [hunk 5]
```diff
@@ -105,8 +105,6 @@ public StudentProfileAttributes getCopy() {
             addNonEmptyError(validator.getInvalidityInfoForNationality(nationality), errors);
         }
 
-        addNonEmptyError(validator.getInvalidityInfoForGender(gender), errors);
```
Comment (xpdavid) [inline src/main/java/teammates/common/datatransfer/attributes/StudentProfileAttributes.java]: not null for gender?

Comment (amrut-prabhu) [reply src/main/java/teammates/common/datatransfer/attributes/StudentProfileAttributes.java]: Added a check for ensuring not null

**Comment by amrut-prabhu** [comment]
I don't think there are any accidental replacements, as can be seen [here](https://github.com/TEAMMATES/teammates/pull/9418/commits/665eb7217d14c2e8725a27324019fce30615dd5f).

**Pull request merged**